src/functions/vacation-request/create-vacation-request/handler.ts

```diff
@@ -0,0 +1,63 @@
+import { middyfy } from "src/libs/lambda";
+import { vacationRequestService } from "src/database/services";
+import { v4 as uuidv4 } from "uuid";
+import type VacationRequestModel from "src/database/models/vacationRequest";
+import type { ValidatedEventAPIGatewayProxyEvent } from "src/libs/api-gateway";
+import type vacationRequestSchema from "src/schema/vacationRequest";
+
+/**
+ * Handler for creating a new vacation request entry in DynamoDB.
+ *
+ * @param event - API Gateway event containing the request body.
+ * @returns Response object with status code
+ */
+export const createVacationRequestHandler: ValidatedEventAPIGatewayProxyEvent<typeof vacationRequestSchema> = async (event) => {
+    if (!event.body) {
+        return {
+            statusCode: 400,
+            body: JSON.stringify({ error: "Request body is required." })
+        };
+    }
+
+    const { personId, draft, startDate, endDate, days, type, message, createdBy, createdAt, updatedAt} = event.body;
+
+    if (!personId || !draft || !startDate || !endDate || !days || !type || !message || !createdBy || !createdAt || !updatedAt) {
```

Comment:
Could be better to separate the user input data checks so that the user knows what exactly is missing from the request. (not an error just keep in mind)